pkg/skaffold/runner/runner.go

```diff
@@ -237,15 +236,34 @@ func (r *SkaffoldRunner) Run(ctx context.Context, out io.Writer, artifacts []*la
 	if err := r.buildTestDeploy(ctx, out, artifacts); err != nil {
 		return err
 	}
-
 	if r.runCtx.Opts.Tail {
 		logger := r.newLogger(out, artifacts)
-		if err := logger.Start(ctx); err != nil {
-			return errors.Wrap(err, "starting logger")
+		return r.TailLogs(ctx, out, logger)
+	}
+	return nil
+}
+
+// Deploy deploys build artifacts.
+func (r *SkaffoldRunner) Deploy(ctx context.Context, out io.Writer, artifacts []build.Artifact) error {
```

Comment:
why do we need both `Deploy()` and `deploy()`? can't we just combine the logic into one method? the `tailLogs` can be conditional right?

Reply:
`runner.Dev` also executes `Deploy` [here](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/runner/dev.go#L132) and it runs its own logging loop.
Its exactly the reason why its repeated here [too](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/runner/runner.go#L241)